controllers/publish-pane.js

```diff
@@ -153,6 +181,21 @@ module.exports = function () {
         });
     },
 
+    onCustomUrlInput: function (e) {
+      var input = e.currentTarget,
+        val = input.value,
+        routes = this.routes;
+
+      // validate that what the user typed in is routable
+      // note: if it's empty string, catch it early (removing custom urls is totally valid)
+      // note: if it's a full url, assume the user knows what they're doing and say it's valid
+      if (val === '' || val.match(/^http/i) || isValidUrl(val, routes)) {
+        input.setCustomValidity('');
+      } else {
+        input.setCustomValidity('Custom URL must match an available route!');
```

Comment:
`customValidity` refers to the message here. Is this setting something that is always a string or can it be another type, like an object?